3p-integrations/gcp/vertex_MaaS/Vertex_tool_calling_for_Llama_4.ipynb

```diff
@@ -0,0 +1,470 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "provenance": []
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# **Why function calling?**\n",
+        "\n",
+        "Imagine asking someone to write down important information without giving them a form or any guidelines on the structure. You might get a beautifully crafted paragraph, but extracting specific details like names, dates, or numbers would be tedious! Similarly, trying to get consistent structured data from a generative text model without function calling can be frustrating. You're stuck explicitly prompting for things like JSON output, often with inconsistent and frustrating results.\n",
+        "\n",
+        "This is where function calling comes in. Instead of hoping for the best in a freeform text response from a generative model, you can define clear functions with specific parameters and data types. These function declarations act as structured guidelines, guiding the Llama model to structure its output in a predictable and usable way. No more parsing text responses for important information!\n",
+        "\n",
+        "Think of it like teaching Llama to speak the language of your applications. Need to retrieve information from a database? Define a search_db function with parameters for search terms. Want to integrate with a weather API? Create a get_weather function that takes a location as input. Function calling bridges the gap between human language and the structured data needed to interact with external systems.\n",
+        "\n",
+        "# **Objectives**\n",
+        "\n",
+        "In this tutorial, you will learn how to use either OpenAI SDK or Vertex AI SDK in Python to make function calls via the Llama 4 Maverick model on Vertex AI. We will use a currency exchange function as an example, you can replace it with another function with the right functionality for you.\n",
+        "This tutorial is based on this Vertex AI codelab: https://codelabs.developers.google.com/codelabs/gemini-function-calling"
+      ],
+      "metadata": {
+        "id": "JIVs8gXB1Auw"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# **First, with OpenAI's SDK**"
```

Comment:
Do you want to add a section on the setup, relevant links required for this on Vertex AI